extensions/amp-a4a/0.1/amp-a4a.js

```diff
@@ -840,18 +847,24 @@ export class AmpA4A extends AMP.BaseElement {
    */
   renderViaSafeFrame_(creativeBody) {
     this.emitLifecycleEvent('renderSafeFrameStart');
-    return utf8Decode(creativeBody).then(creative => {
-      /** @const {!Element} */
-      const iframe = createElementWithAttributes(
-          /** @type {!Document} */(this.element.ownerDocument),
-          'iframe', Object.assign({
-            'height': this.element.getAttribute('height'),
-            'width': this.element.getAttribute('width'),
-            'src': SAFEFRAME_IMPL_PATH + '?n=0',
-            'name': `${SAFEFRAME_VERSION};${creative.length};${creative}`,
-          }, SHARED_IFRAME_PROPERTIES));
-      return this.iframeRenderHelper_(iframe);
+    let decodeErr;
+    const creative = tryUtf8Decode(creativeBody, err => {
```

Comment:
I hated this API with `tryParseJson` and I still hate it. Can't we just have it throw? It'd make this much cleaner:

```js
return new Promise((resolve) => {
  const creative = tryUtf8Decode(creativeBody);
  const iframe = ...;
  return this.iframeRenderHelper_(iframe);
});